### PR TITLE
modlib:need add ctors to init_array and dtors to fini_array

### PR DESCRIPTION
**Pull request by anjiahao1**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary

modlib:need add ctors to init_array and dtors to fini_array

## Impact

modlib

## Testing

esp32 with qemu run module, and pass it

### CODE REVIEW

**Comment by anjiahao1** [comment]
bug fix https://github.com/apache/nuttx/issues/14731
https://github.com/apache/nuttx/pull/13819

**Comment by anjiahao1** [comment]
@tmedicci Could you help me test this PR? Because I passed the test on qemu, I am not sure whether there is still a problem on the real device. Thank you very much

File: libs/libc/modlib/gnu-elf.ld [hunk 1]
```diff
@@ -36,12 +36,14 @@ SECTIONS
   .init_array :
     {
       _sinit = .;
+      *(.ctors)
```
Comment (xiaoxiang781216) [inline libs/libc/modlib/gnu-elf.ld:39]: move after line 40

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by anchao — CHANGES_REQUESTED** [review]
sync the script with other board
https://github.com/apache/nuttx/blob/master/boards/arm/stm32/axoloti/scripts/user-space.ld#L64-L65

**Review by anchao — APPROVED** [review]

**Comment by tmedicci** [comment]
Hi @anjiahao1 , it seems that this PR fixed the issue, but it's working only on QEMU. HW testing is failing (probably) due to https://github.com/apache/nuttx/pull/14625 (I'll check why QEMU and HW testing provide a different result here).

**Pull request merged**